### ISSUE TITLE
Question: watskeburt doesn't work

### ISSUE THREAD

**Issue by Yukon123** [sverweij/watskeburt#49]
### Summary

watskeburt doesn't work with me

![image](https://github.com/user-attachments/assets/f48c942b-9c06-43e8-a4e9-685af2d7a3e5)

I run it for many times and try to switch node version to 18.16.0 ,20.2.0 ,23.3.0   but still get empty result

```bash
$ npx watskeburt HEAD^
^()$
```

- Version used:  4.1.1
- Node version: 18.16.0  20.2.0  23.3.0
- Operating System and version:
- Link to your project:

**Comment by sverweij** [comment]
hi @Yukon123 thanks for raising this issue. 

It seems watskeburt is operating fine, but didn't find any changes - hence an 'empty' regular expression (`^()$`). 

By default the cli with the default reporter (regex) looks for changes in files that have an extension that is in use in the javascript ecosystem (cjs, cjsx, coffee, csx, cts, js, json, jsx, litcoffee, ls, mjs, mts, svelte, ts, tsx, vue, vuex). If between `HEAD^` and the current revision no changes happened on files with these extensions, it will report an 'empty' regular expression.

It's set up that way, because that was the primary use case for watskeburt. 

- The `json` reporter (`--outputType json`) _does_report all extensions.
- Over the past months I've added a (still hidden, experimental) `--extensions` cli option that takes a comma-separated list of extensions (`watskeburt --extensions "md,txt,tex" HEAD^`) - useful if you know in advance which extensions you want to see in the regex it returns.
- For some use cases an option to make the `regex` reporter not filter anything might be useful so that might be a useful future feature (should be pretty straightforward to implement).

**Comment by sverweij** [comment]
FWIW with `watskeburt@4.2.0`, which is just released to npmjs you can pass `--extensions "*"` (or `-e "*"`) to make the default formatter (regex) take _all_ extensions into account.

**Comment by Yukon123** [comment]
![0d4e87d726c90c6311063fb69a56209a02d4d4ab](https://github.com/user-attachments/assets/4dfa6416-24b8-44be-8034-5e02fc9b10e0)

ii add --extensions options and it works which return the right expression that i expect

but as you can see,  current revision has includes the default extension (js)  

The default usage will not find the js file in the revision.  

```
npx watskeburt HEAD^
^()$
```

**Comment by sverweij** [comment]
Hi @Yukon123 thanks for that feedback. The output of that first command looks surprising! It's not

```
^(src/index[.]js$
```

but in stead 
```
^(src/index[.]js
$
```

So there's a carriage return (`\r`) and/ or a linefeed (`\n`) character between the `s` and the `$`. 

Hypothesis: that is because of an issue with parsing the git output on ms-windows systems. 

The end of line (EOL) marker on mswindows (`\r\n`) is different from those in use on most other systems (`\n`). This is why watskeburt uses the `EOL` constant from `node:os`. However, if what `node` thinks is the EOL marker differs from that what `git` thinks it should be (e.g. because of what's in `.gitattributes`) we might run into issues.

To verify that hypothesis:

- I've added a `windows` run on the ci - and the unit tests have [similar issues](https://github.com/sverweij/watskeburt/actions/runs/12136934846) as watskeburt seems to have on your machine.
- PR #51 replaces the use of the `os.EOL` constant with a platform independent regex (`\r?\n`), which seems to alleviate the issue - at least as far as the unit tests go.

I learnt something today: use os.EOL for _writing_, but not for _parsing_ - thanks 🙏 !

**Comment by Yukon123** [comment]
Hi @sverweij Thank you very much for solving this problem that has been bothering me for a long time. 

I upgraded to 4.2.1 and after running the command line, I got the following correct output

![image](https://github.com/user-attachments/assets/acb516cd-bd29-4562-94fa-5108da0100ea)

there is a bug occurs when the API is invoked.

![image](https://github.com/user-attachments/assets/ff806877-dc2a-4658-a019-005b3965b0ec)


This should be caused by `pExtensions` not setting the default value (cjs, cjsx, coffee, csx, cts, js, json, jsx, litcoffee, ls, mjs, mts, svelte, ts, tsx, vue, vuex).

**Comment by sverweij** [comment]
Hi @Yukon123 - yep, passing the list of extensions will definitely help fixing that. I suppose watskeburt could (/ should) also use a 'sensible default' when the `extensions` option is not passed. A.o.t. the long list of javascript-ish extensions the cli uses, that sensible default might be `"*"` for the API, as in the API the caller better knows what extensions are applicable to their context (and maybe even wants to filter themselves on other things than extensions).

Because TypeErrors like the one you encounter shouldn't occur, I'll treat it as a bug that needs fixing - PR coming up.

**Comment by sverweij** [comment]
Small follow-up: you probably already found a workaround (pass `"*"` for extensions, or the comma separated list of the ones you need). Nevertheless I made a little change to the code so when `extensions` isn't passed, no exception is thrown, but instead the default for the API (`"*"`) is taken.

Published as `watskeburt@4.2.2`, just now - that release also expands the `--help` option and the README.md with documenation on the `--extensions` option.

**Comment by Yukon123** [comment]
I really appreciate all your efforts.😃
